src/app/shared/classes/contract.ts

```diff
@@ -652,6 +652,36 @@ export class Contract extends FirebaseDocInterface {
         const docRef = doc(Contract.getCollectionReference(db, company), contractId)
         return onSnapshot(docRef, onNext);
     }
+
+    public getContractedTotalPrice(): number {
+        const total = this.getContractedTotal();
+        return total.get() * this.price.getPricePerUnit(total.getUnit(), total);
```

Comment:
At this point we've both done this quite a few times in the project, it will probably be better to create a method (probably in the price class) to calculate the total price when having a Price and Mass.

Reply:
Added method for getting the total price of a mass in the price class.